README.md

```diff
@@ -1,129 +1,170 @@
+
+For debugging purposes, make sure there is a Nucleo plugged into your computer and connected to the leaderboard via JTAG. Run ```openocd``` in one bash session and ```gdb-multiarch``` in another, then attach to the process openocd has started.
+
+**[Back to top](#table-of-contents)**
+
+## Release Process
+
+Each release commit will be tagged with the version number (**MX.X.X**) and live on a production branch (**production/MX.X**). The controls-leader.elf binary corresponding to the release will also be uploaded to our [Releases](https://github.com/lhr-solar/Controls/releases) page on Github.
+
+### Versioning
+
+Our versioning system does NOT follow semantic versioning, but instead is based around iterations of the Longhorn Racing Solar Car. Each major version change corresponds to a different iteration of the car for a different year of competition.
+
+Current versions:
+- **M1** is for the FSGP 2022 Solar Car
+- **M2** is for the FSGP 2024 Solar Car
+- **M3** is for the FSGP 2025 Solar Car
+
+Intermediate version names usually refers to a major change that exists for the same vehicle, such as breaking API changes or a rearchitecture.
+
+**[Back to top](#table-of-contents)**
+
+## Contributing
+
+### Branch Convention
+
+**Feature** branches (```feature/*-###```) are for features or bug fixes that are specifically associated with an existing issue ticket.
+- Feature branches end with the issue ticket number that the feature is solving.
+
+**Development** branches (```dev/*```) are for temporary development purposes or hotfixes.
+- If a feature, bug, or enhancement starts being worked on without an existing issue ticket, a development branch can be created for it. 
+- Hotfixes without associated issue tickets can also be created on development branches and merged from a PR without the need for a feature branch, given that the change is small.
+- Any other experimentation or temporary needs can live on a development branch.
+
+**Integration** branches (```integration/*-###```) are for integration testing.
+- This includes integration between Controls and other systems on the vehicle, as well as integration between our internal Applications/Drivers.
+- Integration branches end with the issue ticket number for the specific integration being done.
+
+**Production** branches (```production/MX.X```) are for specific versions of the codebase. ```master``` will always host the latest version.
+
+### Review Process
+
+Commit frequently into your branch. Create a Pull Request whenever you are ready to add you working code to the master branch. You must select 2 reviewers for approval. Any trained contributors are allowed to review and approve code. See our [PR Template](#PULL_REQUEST_TEMPLATE.md) for more review guidelines.
+
+**[Back to top](#table-of-contents)**
 
```

Comment:
Maybe add a "Current Work Direction" section with a description of what the current status / progress is.

Reply:
Could also have a clickup link there.

Reply:
Added in the Project Status section.